variation-commons-core/src/main/java/uk/ac/ebi/eva/commons/core/models/factories/VariantVcfFactory.java

```diff
@@ -351,4 +361,38 @@ protected static int mapToMultiallelicIndex(int parsedAllele, int numAllele) {
         }
         return correctedAllele;
     }
+
+    private void checkVariantInformation(Variant variant, String fileId,
+                                           String studyId) throws NonVariantException, IncompleteInformationException {
+
+        // check genotypes
+        VariantSourceEntry variantSourceEntry = variant.getSourceEntry(fileId, studyId);
+        List<Map<String, String>> samplesData = variantSourceEntry.getSamplesData();
+        if (!samplesData.isEmpty()) {
+            if (samplesData.stream().map(m -> m.get("GT")).noneMatch(this::genotypeHasAlternateAllele)) {
+                throw new NonVariantException("TODO MESSAGE HERE");
+            }
+        } else {
+            // check info
+            boolean hasAC = checkAttributeIsNotZero(variantSourceEntry, "AC");
+            boolean hasAF = checkAttributeIsNotZero(variantSourceEntry, "AF");
+            boolean hasAN =  checkAttributeIsNotZero(variantSourceEntry, "AN");
+            boolean hasGTC =  checkAttributeIsNotZero(variantSourceEntry, "GTC");
+            if (!hasAC && !hasAF && !hasAN && !hasGTC) {
+                throw new IncompleteInformationException(variant);
+            }
+        }
+    }
+
+    private boolean genotypeHasAlternateAllele(String sampleField) {
+        return Arrays.stream(sampleField.split("[/|]")).anyMatch(allele -> allele.equals("1"));
```

Comment:
maybe it's good to document here that we shouldn't check for "2"s because this is done after processing the indexes in multiallelic variants

Reply:
👍

Reply:
Agree, it is unclear otherwise.